packages/server/src/Connection.ts

```diff
@@ -163,14 +163,13 @@ export class Connection {
 
       if (this.document.hasConnection(this)) {
         this.document.removeConnection(this)
-        clearInterval(this.pingInterval)
+      }
 
-        this.webSocket.removeListener('close', this.boundClose)
-        this.webSocket.removeListener('message', this.boundHandleMessage)
-        this.webSocket.removeListener('pong', this.boundHandlePong)
+      this.webSocket.removeListener('close', this.boundClose)
+      this.webSocket.removeListener('message', this.boundHandleMessage)
+      this.webSocket.removeListener('pong', this.boundHandlePong)
 
-        this.callbacks.onClose.forEach((callback: (arg0: Document, arg1?: CloseEvent) => any) => callback(this.document, event))
-      }
+      this.callbacks.onClose.forEach((callback: (arg0: Document, arg1?: CloseEvent) => any) => callback(this.document, event))
```

Comment:
There's an open question in our minds about whether this callbacks.onClose is intended to be conditional on the document.hasConnection... we think the callbacks should always fire when the connection is closed?

Reply:
I moved this back inside the condition as it seems required to pass tests; however, I didn't look into it further.